testutil/network/network.go

```diff
@@ -1,133 +1,795 @@
 package network
```

Comment:
this looks like copied code from cosmos-sdk with is generally a bad idea. if for example we ever update the cosmos-sdk version we would need to update all the copied code as well, which normally not happens

Reply:
@jmastr I discussed this with @LaurentMontBlanc. We aligned that we want to move ahead with this approach so that we can configure our testing network in an easy way in the future, we might also be able to reduce the complexity a bit over time.